### PR TITLE
iOS: Add a maxFontSizeMultiplier prop to <Text> and <TextInput>

### PR DESCRIPTION
**Pull request by rigdern**
**Motivation**

Whenever a user changes the system font size to its maximum allowable setting, React Native apps that allow font scaling can become unusable because the text gets too big. Experimenting with a native app like iMessage on iOS, the font size used for non-body text (e.g. header, navigational elements) is capped while the body text (e.g. text in the message bubbles) is allowed to grow.

This PR introduces a new prop on `<Text>` and `<TextInput>` called `maxFontSizeMultiplier`. This enables devs to set the maximum allowed text scale factor on a Text/TextInput. The default is 0 which means no limit.

Another PR will add this feature to Android.

**Test Plan**

I created a test app which utilizes all categories of values of `maxFontSizeMultiplier`:
  - `undefined`: inherit from parent
  - `0`: no limit
  - `1`, `1.2`: fixed limits

I tried this with `Text`, `TextInput` with `value`, and `TextInput` with children. For `Text`, I also verified that nesting works properly (if a child `Text` doesn't specify `maxFontSizeMultiplier`, it inherits it from its parent).

Lastly, we've been using a version of this in Skype for several months.

**Release Notes**

[GENERAL] [ENHANCEMENT] [Text/TextInput] - Added maxFontSizeMultiplier prop to prevent some text from getting unusably large as user increases OS's font scale setting (iOS)

Adam Comella
Microsoft Corp.

### CODE REVIEW

**Comment by janicduplessis** [comment]
@rigdern I think it would be better to call it `maxFontSizeMultiplier` for consistency with `fontSizeMultiplier` and `allowFontScaling`.

File: Libraries/Text/Text/RCTTextViewManager.m [hunk 1]
```diff
@@ -24,7 +24,6 @@ @interface RCTTextViewManager () <RCTUIManagerObserver>
 @implementation RCTTextViewManager
 {
   NSHashTable<RCTTextShadowView *> *_shadowViews;
-  CGFloat _fontSizeMultiplier;
```
Comment (shergin) [inline Libraries/Text/Text/RCTTextViewManager.m]: 👍

**Comment by shergin** [comment]
@rigdern Can/should we rename this to `maxFontSizeMultiplier` though?

**Comment by rigdern** [comment]
@shergin, @janicduplessis I renamed the prop to `maxFontSizeMultiplier` as suggested.

Thanks for the feedback.

**Pull request closed without merging**